src/main/java/Game.java

```diff
@@ -20,21 +25,59 @@ public Game(@NotNull Board board,@NotNull Player player1,@NotNull Player player2
     public void start() {
         board.clearBoard();
         gameStatus = GameStatus.STARTED;
-        while (gameStatus == GameStatus.STARTED) {
-            turn(whitePlayer);
-            if(gameStatus == GameStatus.FINISHED) break;
-            turn(blackPlayer);
+        while (gameStatus != GameStatus.FINISHED) {
+            turn();
         }
         end();
     }
 
-    public void turn(@NotNull Player player) {
-        if(board.hasMoreThanOneFreeCell()) {
-            player.putStone(board);
-        } else {
-            player.lastMove(board);
-            gameStatus = GameStatus.FINISHED;
+    public void turn() {
+        Player currentPlayer = nextPlayer();
+        Position position = null;
+        checkCurrentGameStatus();
+        switch (gameStatus) {
+            case FREEDOM:
+                position = getPositionWithFreedom(currentPlayer);
+                break;
+            case NO_FREEDOM:
+                position = getPositionWithNoFreedom(currentPlayer);
+                break;
+            case LAST_MOVE:
+                position = playLastMove(currentPlayer);
+                gameStatus = GameStatus.FINISHED;
+                break;
         }
+        allPlayersMoves.add(new Move(currentPlayer, position));
+    }
+
+    @NotNull
+    private Player nextPlayer() {
+        try {
+            Player previousPlayer = allPlayersMoves.getLast().getPlayer();
+            if (previousPlayer.equals(whitePlayer)) return blackPlayer;
```

Comment:
Keeping track of all the players' moves may be unnecessary (a simple boolean field would work to know which was the last player), but we can just keep it for now

Reply:
We have to know the position of the last move. So, instead of the linked list, we can have a `Move lastMove` field